app/MMB/src/navigation/DetailScreen.js

```diff
@@ -141,25 +140,29 @@ export default class DetailsScreen extends React.Component<Props, State> {
     ).screen;
 
     return (
-      <BookingDetailContext.Provider
-        isPastBooking={this.props.isPastBooking}
-        bookingId={this.props.bookingId}
-      >
-        <Layout
-          menuComponent={
-            <AdaptableLayout.Consumer
-              renderOnWide={<MainMenu openMenu={this.changeContentOnTablet} />}
-              renderOnNarrow={<MainMenu openMenu={this.openMenuOnMobile} />}
-            />
-          }
-          containerComponent={
-            <ContainerComponent
-              navigation={this.props.navigation}
-              bookingId={this.props.bookingId}
-            />
-          }
-        />
-      </BookingDetailContext.Provider>
+      <Layout
+        menuComponent={
+          <AdaptableLayout.Consumer
+            renderOnWide={<MainMenu openMenu={this.changeContentOnTablet} />}
+            renderOnNarrow={<MainMenu openMenu={this.openMenuOnMobile} />}
+          />
+        }
+        containerComponent={
+          <ContainerComponent navigation={this.props.navigation} />
+        }
+      />
     );
   };
 }
+
+type Props = {|
+  +navigation: NavigationType,
+|};
+
+export default function DetailScreenWithContext(props: Props) {
+  return (
+    <BookingDetailContext.Consumer>
+      {({ bookingId }) => <DetailsScreen {...props} bookingId={bookingId} />}
```

Comment:
Is it necessary? I see it's only in `openMenuOnMobile` so maybe we don't need it here at all and we can consume it directly where needed? :)